bin/migrate-oats-data/documents/post_launch/alcs_documents_to_notification_documents.py

```diff
@@ -0,0 +1,138 @@
+
+                    last_document_id = dict(rows[-1])["oats_document_id"]
+                    successful_inserts_count = (
+                        successful_inserts_count + documents_to_be_inserted_count
+                    )
+
+                    logger.debug(
+                        f"retrieved/inserted items count: {documents_to_be_inserted_count}; total successfully inserted/updated documents so far {successful_inserts_count}; last inserted oats_document_id: {last_document_id}"
+                    )
+                except Exception as e:
+                    conn.rollback()
+                    logger.exception(f"Error {e}")
+                    failed_inserts_count += len(rows)
+                    last_document_id = last_document_id + 1
+
+    logger.info(f"Total amount of successful inserts: {successful_inserts_count}")
+    logger.info(f"Total amount of failed inserts: {failed_inserts_count}")
+
+
+def _insert_records(conn, cursor, rows):
+    number_of_rows_to_insert = len(rows)
+
+    if number_of_rows_to_insert > 0:
+        insert_query = _compile_insert_query(number_of_rows_to_insert)
+        rows_to_insert = _prepare_data_to_insert(rows)
+        cursor.execute(insert_query, rows_to_insert)
+        conn.commit()
+
+
+def _compile_insert_query(number_of_rows_to_insert):
+    documents_to_insert = ",".join(["%s"] * number_of_rows_to_insert)
+    return f"""
+                        INSERT INTO alcs.notification_document(
+                            notification_uuid,
+                            document_uuid,
+                            type_code,
+                            visibility_flags,
+                            oats_document_id,
+                            oats_application_id,
+                            audit_created_by           
+                        )
+                        VALUES{documents_to_insert}
+                        ON CONFLICT (oats_document_id, oats_application_id) DO UPDATE SET 
+                            notification_uuid = EXCLUDED.notification_uuid, 
+                            document_uuid = EXCLUDED.document_uuid, 
+                            type_code = EXCLUDED.type_code,
+                            visibility_flags = EXCLUDED.visibility_flags,
+                            audit_created_by = EXCLUDED.audit_created_by;
+    """
+
+
+def _prepare_data_to_insert(rows):
+    row_without_last_element = []
+    for row in rows:
+        mapped_row = _map_data(row)
+        row_without_last_element.append(tuple(mapped_row.values()))
+
+    return row_without_last_element
+
+
+def _map_data(row):
+    return {
+        "notification_uuid": row["notification_uuid"],
+        "document_uuid": row["document_uuid"],
+        "type_code": row["type_code"],
+        "visibility_flags": row["visibility_flags"],
+        "oats_document_id": row["oats_document_id"],
+        "oats_application_id": row["oats_application_id"],
+        "audit_created_by": OATS_ETL_USER,
+    }
```

Comment:
I do not see "survey_plan_number." Is it going to be in a separate ticket?

Reply:
updated